shell/src/main/java/net/corda/tools/shell/FlowShellCommand.java

```diff
@@ -185,4 +186,106 @@ public void retryAllPausedHospitalized() {
             out.println("One or more paused hospitalized flows failed to retry.", Decoration.bold, Color.red);
         }
     }
+
+    @Command
+    @Usage("Recover a finality flow by flow uuid.")
+    public void recoverFinality(
+            @Usage("The UUID for the flow that we wish to recover") @Argument String id,
+            @Usage("Flag to force recovery of flows that are in a HOSPITALIZED or PAUSED state.")
+            @Option(names = { "f", "force-recover" }) Boolean forceRecover
+    ) {
+        logger.info("Executing command \"flow recoverFinality {}\".", id);
+        StateMachineRunId parsedId = parseStateMachineRunId(id, out, objectMapper(null));
+
+        Boolean recoveryResult;
+        if (forceRecover != null)
+            recoveryResult = getFlowRPCShellCommand().recoverFinalityFlow(parsedId, forceRecover);
+        else
+            recoveryResult = getFlowRPCShellCommand().recoverFinalityFlow(parsedId);
+
+        if (recoveryResult) {
+            out.println("Recovered finality flow " + parsedId, Decoration.bold, Color.yellow);
+        } else {
+            out.println("Failed to recover finality flow " + parsedId, Decoration.bold, Color.red);
```

Comment:
Related to my comment in other PR (https://github.com/corda/enterprise/pull/4655), does this also mean failed to recover as recovery not needed.

Reply:
Good point - need to differentiate between "failed to recover" and "recovery not required"

Reply:
https://r3-cev.atlassian.net/browse/ENT-9965